src/modules/feedback/components/FeedbackBox.js

```diff
@@ -73,9 +75,10 @@ export class FeedbackBox extends React.PureComponent<PropsType> {
           commentMessage={isPositiveRatingSelected ? t('positiveComment') : t('negativeComment')}
           onCommentChanged={onCommentChanged}
           required={!isPositiveRatingSelected} />
+        <Description>{requestError ? t('failedSendingFeedback') : null}</Description>
         <TextButton
           disabled={!isPositiveRatingSelected && !comment}
-          onClick={onSubmit}
+          onClick={requestError ? null : onSubmit}
```

Comment:
This does not seem right (especially in terms of ui/ux, but also where this is handled). You shouldn't just disable a button without giving the user a hint that it is disabled and providing an option to enable it again. Imo the logic to show that sending the feedback failed should not reside here, but instead either in the refactored `FeedbackThanksMessage` (possibly with a try again button) or a entire new component, both rendered in the `FeedbackModal`. If you use the suggestion of the enum, you could switch between its value to render either a `FeedbackBox`, a `FeedbackThanksMessage` or a `FeedbackError` there.

Reply:
changed